File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/movies/GetUserWatchList.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.repository.MoviesRepository;
import com.moovie.backend.domain.model.MovieListing;
import org.springframework.stereotype.Service;

@Service
public class GetUserWatchList {

    private final MoviesRepository moviesRepository;

    public GetUserWatchList(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieListing invoke(String userId, int page) {
        return moviesRepository.getUserWatchList(userId, page);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/GetUserWatchList.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.repository.MoviesRepository
import org.mobyle.model.MovieListing

class GetUserWatchList(private val repository: MoviesRepository) {
    operator fun invoke(userId: String, page: Int): MovieListing = runBlocking {
        repository.getUserWatchList(userId, page)
    }
}
